lib/interceptor/cache.js

```diff
@@ -0,0 +1,167 @@
+'use strict'
+
+const CacheHandler = require('../handler/cache-handler')
+const MemoryCacheStore = require('../cache/memory-cache-store')
+const CacheRevalidationHandler = require('../handler/cache-revalidation-handler')
+
+/**
+ * Gives the downstream handler the request's cached response or dispatches
+ *  it if it isn't cached
+ * @param {import('../../types/cache-interceptor.d.ts').default.CacheOptions} globalOpts
+ * @param {*} dispatch TODO type
+ * @param {import('../../types/dispatcher.d.ts').default.RequestOptions} opts
+ * @param {import('../../types/dispatcher.d.ts').default.DispatchHandlers} handler
+ * @param {import('../../types/cache-interceptor.d.ts').default.CacheStoreValue | undefined} value
+ */
+function handleCachedResult (
+  globalOpts,
+  dispatch,
+  opts,
+  handler,
+  value
+) {
+  if (!value) {
+    // Request isn't cached, let's continue dispatching it
+    dispatch(opts, new CacheHandler(globalOpts, opts, handler))
+    return
+  }
+
+  // Dump body on error
+  opts.body?.on('error', () => {}).resume()
+
+  const respondWithCachedValue = () => {
+    const ac = new AbortController()
+    const signal = ac.signal
+
+    try {
+      handler.onConnect(ac.abort)
+      signal.throwIfAborted()
+
+      // Add the age header
+      // https://www.rfc-editor.org/rfc/rfc9111.html#name-age
+      const age = Math.round((Date.now() - value.cachedAt) / 1000)
+
+      // Copy the headers in case we got this from an in-memory store. We don't
+      //  want to modify the original response.
+      const headers = [...value.rawHeaders]
+      headers.push(Buffer.from('age'), Buffer.from(`${age}`))
```

Comment:
Why create a buffer from age every time? 

create a buffer at the top and reuse it?!